### DISCUSSION TITLE
Graphql api route using localhost instead of NEXT_PUBLIC_ when deploying to vercel

### DISCUSSION THREAD

**Discussion by EthanNC** [vercel/next.js#19157] [Help]
I'm trying to deploy a fullstack graphql app on vercel. Ive gotten the project to deploy but for some reason my api routes are still calling ``` https://localhost:3000``` when I think they should be calling my ```NEXT_PUBLIC_...``` environment variable.
Based on the logs I think it would be the client side issue, but I am not sure. Any insight would be most valuable.
Here are the function logs: 
<img width="1320" alt="Screen Shot 2020-11-13 at 5 05 16 PM" src="https://user-images.githubusercontent.com/14959322/99126518-55572100-25d4-11eb-9b57-08d21ab656e6.png">

**Comment by waptik** [comment]
If you are calling the api with `process.env.NEXT_PUBLIC_...`, then did you set value for `NEXT_PUBLIC_...` env variable on vercel for both preview and production deployments or inside  `.env(*)` if you've pushed git or vercel?
